### PR TITLE
feat(server/http): add server

### PR DESCRIPTION
**Pull request by Davincible**

### CODE REVIEW

**Comment by Davincible** [comment]
We need to add the logger here

File: server/http/router/router/pluings.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package router
```
Comment (jochumdev) [inline server/http/router/router/pluings.go]: Typo in filename

File: server/http/router/router/router.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+// NewRouterFunc can be used to create a new router.
+type NewRouterFunc func() Router
+
+// TODO: check all comments
+
+// The Router interface as found here is a copy of the Chi router interface.
+// It was selected for it's simplicity compliance with the stdlib.
+// Some methods were removded as they were considered unnecessary for go-micro.
```
Comment (jochumdev) [inline server/http/router/router/router.go]: removed

File: server/http/utils/ip/ip.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
+// Package ip provides validation and parsing utilities.
```
Comment (jochumdev) [inline server/http/utils/ip/ip.go:1]: How does this work with ipv6 ? Seems not?

File: server/http/utils/tls/tls.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+	config := tls.Config{
+		Certificates: certs,
+		MinVersion:   tls.VersionTLS13,
+	}
+
+	return &config, nil
+}
+
+// GenTlSConfig creates a self signed certificate and returns it in a TSL config.
+func GenTlSConfig(addr ...string) (*tls.Config, error) {
```
Comment (jochumdev) [inline server/http/utils/tls/tls.go]: Shouldn't it be GenTLSConfig?

File: server/http/utils/wg/wg.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
+// Package wg implements a double waitgroup that can be used to keep track
```
Comment (jochumdev) [inline server/http/utils/wg/wg.go]: This is very "clever" can you explain more what this is for?

File: server/http/utils/wg/wg.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+// of a global waitgroup at the same time. Adding and stubstracting from
+// the waitgroup will impact both the global and local waiggroups, but waiting
```
Comment (jochumdev) [inline server/http/utils/wg/wg.go]: Typo waitgroups

File: server/tests/proto/generate.go [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package proto
+
+// Download Google proto HTTP annotation libs
+//go:generate wget -q -O google/api/annotations.proto https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/annotations.proto
```
Comment (jochumdev) [inline server/tests/proto/generate.go:6]: May switch from wget to curl? wget is not every installed while curl mostly is.

**Review by jochumdev — CHANGES_REQUESTED** [review]

File: server/http/utils/wg/wg.go (hunk 5 above)
Comment (Davincible) [reply server/http/utils/wg/wg.go]: Not my idea. It's already a v4 util. I really like it as well. I'm not sure if its at place here in the HTTP package, but should be used somewhere.

It allows you to add to a runtime used globally, but only wait for your added routines, and not all other routines, while at the same time giving the safety that the global process knows about your routines.

File: server/tests/proto/generate.go (hunk 7 above)
Comment (Davincible) [reply server/tests/proto/generate.go:6]: We could but it's not important, actually only needs to be done once, which it has been, just put it there for documentation

File: server/http/utils/wg/wg.go (hunk 5 above)
Comment (jochumdev) [reply server/http/utils/wg/wg.go]: So if it's not used you can delete it, right?

File: server/http/utils/ip/ip.go (hunk 3 above)
Comment (Davincible) [reply server/http/utils/ip/ip.go:1]: Added tests

**Pull request merged**